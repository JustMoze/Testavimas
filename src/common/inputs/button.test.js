import React from 'react'
import { shallow } from 'enzyme'
import Button from "./button";

import {configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json'
import renderer from 'react-test-renderer'


configure({adapter: new Adapter()});

describe('Button test', () => {
  const onClick = jest.fn();
  const props = {
    facebook: true,
    onClick: onClick
  }
  const wrapper = shallow(<Button {...props} />)
  it('Renders footer correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  // check if renders one buttons
  expect(wrapper.find('button')).toHaveLength(1);
})
