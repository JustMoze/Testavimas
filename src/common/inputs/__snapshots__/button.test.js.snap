// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test Renders footer correctly 1`] = `
<Fragment>
  <styled.div
    onClick={[MockFunction]}
  >
    <button>
      <div>
        <FacebookIcon />
        Facebook
      </div>
    </button>
  </styled.div>
</Fragment>
`;
